### ISSUE TITLE
Should we remove "precooked" entities

### ISSUE THREAD

**Issue by NicolasAnquetil** [moosetechnology/Moose#2161]
Precooked entities are TClass, TMethod, ...

That propose a standard combination of traits for some common entities
The problem is that we use them and often add again traits that they already have.
This causes traits conflicts.

So either we clean them and describe them and make sure everything is corret
Or we simply remove them and recreate everything by hand for all models (which we are already doing actully which causes the traits conflicts)

**Comment by sbragagnolo** [comment]
I completely agree with this removing. When doing the Csharp model i felt really upset with the redundance of the traits. Maybe we could move this entities to an other package, and load it on demand, like that we can still being compatible. If you agree i can get my hands on this. If not i would need some for thinking on a way to migrate existing projects to the model without this traits, since i am not really sure about how to propose this kind of structural migraiton for the users.
